lib/Dredd.js

```diff
@@ -40,215 +99,79 @@ class Dredd {
       end: 0,
       duration: 0,
     };
-    this.files = [];
-    this.transactions = [];
-    this.runner = new Runner(this.configuration);
+    this.transactionRunner = new TransactionRunner(this.configuration);
     this.logger = logger;
-    configureReporters(this.configuration, this.stats, this.tests, this.runner);
-    this.logProxySettings();
   }
 
-  logProxySettings() {
-    const proxySettings = [];
-    for (const envVariableName of Object.keys(process.env || {})) {
-      const envVariableValue = process.env[envVariableName];
-      if (!Array.from(PROXY_ENV_VARIABLES).includes(envVariableName.toUpperCase())) { continue; }
-      if (envVariableValue === '') { continue; }
-      proxySettings.push(`${envVariableName}=${envVariableValue}`);
+  prepareAPIdescriptions(callback) {
```

Comment:
The prepareAPIdescriptions() method isn't a real method of a real object as the Dredd class doesn't need to be a class in the first place. I created it just to make run() shorter. It roughly contains the pipeline turning API description locations into API descriptions, then parsing them, and then compiling them into transactions. Once we have async/await/promises and the code is less tied with 'this.configuration', it might make sense to refactor this further.